code/game/machinery/suit_storage_unit.dm

```diff
@@ -1,184 +1,406 @@
+	else if(occupant)
+		add_overlay("human")
+
+/obj/machinery/suit_storage_unit/attackby(obj/item/I as obj, mob/user as mob, params)
+	if(!is_operational())
+		return
+	if(isscrewdriver(I))
+		panel_open = !panel_open
+		playsound(loc, I.usesound, 100, 1)
+		to_chat(user, text("<span class='notice'>You [panel_open ? "open up" : "close"] the unit's maintenance panel.</span>"))
+		updateUsrDialog()
 
 /obj/machinery/suit_storage_unit/power_change()
-	if( powered() )
-		src.ispowered = 1
-		stat &= ~NOPOWER
-		src.update_icon()
+	..()
+	if(!is_operational() && state_open)
+		open_machine()
+		dump_contents()
+	update_icon()
+
+
+/obj/machinery/suit_storage_unit/proc/dump_contents()
+	dropContents()
+	helmet = null
+	suit = null
+	mask = null
+	storage = null
+	occupant = null
+
+/obj/machinery/suit_storage_unit/deconstruct(disassembled = TRUE)
+	open_machine()
+	dump_contents()
+	new /obj/item/stack/sheet/metal (loc, 2)
+	qdel(src)
+
+/obj/machinery/suit_storage_unit/MouseDrop_T(atom/A, mob/user)
+	if(user.stat || user.lying || !Adjacent(user) || !Adjacent(A) || !isliving(A))
+		return
+	var/mob/living/target = A
+	if(!state_open)
+		to_chat(user, "<span class='warning'>The unit's doors are shut!</span>")
+		return
+	if(!is_operational())
+		to_chat(user, "<span class='warning'>The unit is not operational!</span>")
+		return
+	if(occupant || helmet || suit || storage)
+		to_chat(user, "<span class='warning'>It's too cluttered inside to fit in!</span>")
+		return
+
+	if(target == user)
+		user.visible_message("<span class='warning'>[user] starts squeezing into [src]!</span>", "<span class='notice'>You start working your way into [src]...</span>")
 	else
-		spawn(rand(0, 15))
-			src.ispowered = 0
-			stat |= NOPOWER
-			src.islocked = 0
-			src.isopen = 1
-			src.dump_everything()
-			src.update_icon()
-
-
-/obj/machinery/suit_storage_unit/ex_act(severity)
-	switch(severity)
-		if(1.0)
-			if(prob(50))
-				src.dump_everything() //So suits dont survive all the time
-			qdel(src)
-			return
-		if(2.0)
-			if(prob(50))
-				src.dump_everything()
-				qdel(src)
+		target.visible_message("<span class='warning'>[user] starts shoving [target] into [src]!</span>", "<span class='userdanger'>[user] starts shoving you into [src]!</span>")
+
+	if(do_mob(user, target, 30))
+		if(occupant || helmet || suit || storage)
 			return
+		if(target == user)
+			user.visible_message("<span class='warning'>[user] slips into [src] and closes the door behind [user.p_them()]!</span>", "<span class=notice'>You slip into [src]'s cramped space and shut its door.</span>")
 		else
+			target.visible_message("<span class='warning'>[user] pushes [target] into [src] and shuts its door!<span>", "<span class='userdanger'>[user] shoves you into [src] and shuts the door!</span>")
+		close_machine(target)
+		add_fingerprint(user)
+
+/obj/machinery/suit_storage_unit/proc/cook()
+	if(uv_cycles)
+		uv_cycles--
+		uv = TRUE
+		locked = TRUE
+		update_icon()
+		if(occupant)

```

Comment:
I'm not sure where it'll go, but you may need a check here (and elsewhere) to check if the occupant's actually inside the SSU before doing anything. Weirdness could arise with crap like bluespace crystal teleportation 😄